massa-pos-exports/src/pos_final_state.rs

```diff
@@ -999,18 +991,15 @@ impl PoSFinalState {
             let db = self.db.read();
 
             while let Some((serialized_key, _)) = match found_cycles.last() {
-                Some(prev_cycle) => {
-                    let cycle_prefix = self.cycle_history_cycle_prefix(*prev_cycle);
-
-                    db.iterator_cf(
+                Some(prev_cycle) => db
+                    .iterator_cf(
                         STATE_CF,
                         MassaIteratorMode::From(
-                            &upper_limit_prefix!(cycle_prefix),
+                            &self.cycle_history_cycle_prefix(prev_cycle.saturating_add(1)),
```

Comment:
what is this ?

Reply:
This basically reverts the previous fix (that avoided the assumption that cycles are in order in the DB) made in https://github.com/massalabs/massa/pull/4104.

This is not needed, but it simplifies a bit the code.

Reply:
do we have tests to ensure that the change won't cause a regression ?

Reply:
test_pos_cache_recomputation() at the bottom of the file wil catch this. It was the test I added in #4104 to make sure the cycle_history is correctly recovered from the db (in the correct order)